src/hmem_ze.c

```diff
@@ -583,11 +643,16 @@ int ze_hmem_init(void)
 	if (ze_ret)
 		goto err;
 
-	ret = ze_hmem_init_fds();
+	ze_ret = ze_hmem_init_fds();
 	if (ret)
 		goto err;
 
 	for (num_devices = 0; num_devices < count; num_devices++) {
+		ze_ret = ze_hmem_find_copy_only_engine(num_devices, &ordinals[num_devices]);
+		if (ze_ret || ordinals[num_devices] == -1)
+			goto err;
```

Comment:
Does copy only engine always exist? On failure, would it better to use a mixed engine than to fail?

Reply:
Good point, yes we should definitely fall back to a mixed engine.